bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -269,6 +272,13 @@ public int open () {
 	return userResponse;
 }
 
+private NSString getLabelForButton(int buttonId, String messageId) {
+	if (labels !=null && labels.containsKey(buttonId)) {
+		return NSString.stringWith(labels.get(buttonId));
```

Comment:
API looks good. It'll be good to have a null check here in case returned label is null.

Reply:
There is already null check and one can set label to anything incl. null, empty string, any whitespace character and etc. and all would not make sense . We are also not making any such check in e.g. Button.setText

Reply:
If labels.get(buttonId) returns null, NSString.stringWith() results in NPE. In case of Button.setText, it doesn't accept null string.